src/pyspiffe/svid/jwt_svid_validator.py

```diff
@@ -0,0 +1,132 @@
+    See `SPIFFE JWT-SVID standard <https://github.com/spiffe/spiffe/blob/master/standards/JWT-SVID.md>`
+
+    """
+
+    _REQUIRED_CLAIMS = ['aud', 'exp', 'sub']
+    _SUPPORTED_ALGORITHMS = [
+        'RS256',
+        'RS384',
+        'RS512',
+        'ES256',
+        'ES384',
+        'ES512',
+        'PS256',
+        'PS384',
+        'PS512',
+    ]
+    _SUPPORTED_TYPES = ['JWT', 'JOSE']
+
+    def __init__(self) -> None:
+        pass
+
+    def validate_header(self, header: Dict) -> None:
+        """Validates token header by verifing if header specifies supported algortihms and token type. Type is optional but in case it is present, it must be set to the supported values.
+
+        Args:
+            header: token header.
+
+        Returns:
+            None.
+
+        Raises:
+            ValueError: in case header is not specified.
+            InvalidAlgorithmError: in case specified 'alg' is not supported as specified by the SPIFFE standard.
+            InvalidTypeError: in case 'typ' is present in header but is not set to 'JWT' or 'JOSE'.
+        """
+        if not header:
+            raise ValueError(INVALID_INPUT_ERROR.format('header alg cannot be empty'))
+
+        alg = header.get('alg')
+        if not alg:
+            raise ValueError(INVALID_INPUT_ERROR.format('header alg cannot be empty'))
+
+        if alg not in self._SUPPORTED_ALGORITHMS:
+            raise InvalidAlgorithmError(alg)
+
+        typ = header.get('typ')
+        if typ and typ not in self._SUPPORTED_TYPES:
+            raise InvalidTypeError(typ)
+
+    def validate_claims(self, payload: Dict, expected_audience: List) -> None:
+        """Validates payload for required claims (aud, exp, sub).
+
+        Args:
+            payload: token playload.
+            expected_audience: audience as a list of strings used to validate the 'aud' claim.
+
+        Returns:
+            None
+
+        Raises:
+            InvalidClaimError: in case a required claim is not present in payload or expected_audience is not a subset of audience_claim.
+            TokenExpiredError: in case token is expired.
+            ValueError: in case expected_audience is empty.
+        """
+        for claim in self._REQUIRED_CLAIMS:
+            if not payload.get(claim):
+                raise InvalidClaimError(claim)
+        self._validate_exp(payload.get('exp'))
+        self._validate_aud(payload.get('aud'), expected_audience)
+
+    def _validate_exp(self, expiration_date: str) -> None:
+        """Verifies expiration.
+        Note: If and when https://github.com/jpadilla/pyjwt/issues/599 is fixed, this can be simplified/removed.
+
+        Args:
+            expiration_date: date to check if it is expired.
+
+        Raises:
+            TokenExpiredError: in case it is expired.
+            InvalidClaimError: in case expiration_date is not provided.
+        """
+        if not expiration_date or expiration_date == '':
```

Comment:
nit: `if not expiration_date` should already cover the scenario where `expiration_date == ''` because strings are [falsy](https://docs.python.org/3.6/library/stdtypes.html#truth-value-testing) in Python

Reply:
Done.